apps/src/aichat/views/assets/FilePreview.tsx

```diff
@@ -0,0 +1,104 @@
+import FontAwesomeV6Icon from '@code-dot-org/component-library/fontAwesomeV6Icon';
+import {WithTooltip} from '@code-dot-org/component-library/tooltip';
+import {StrongText} from '@code-dot-org/component-library/typography';
+import classNames from 'classnames';
+import React, {useEffect, useRef, useState} from 'react';
+
+import aichatI18n from '@cdo/apps/aichat/locale';
+
+import styles from './staged-files-preview.module.scss';
+
+const FilePreview: React.FC<{
+  type: 'pdf' | 'image';
+  filename: string;
+  url: string;
+  isUploading?: boolean;
+  onRemove?: () => void;
+}> = ({type, filename, url, isUploading, onRemove}) => {
+  const imageRef = useRef<HTMLImageElement>(null);
+  const [imageLoaded, setImageLoaded] = useState<boolean>(false);
+
+  useEffect(() => {
+    const imageElement = imageRef.current;
+    if (!imageElement) {
+      return;
+    }
+
+    const handleLoad = () => {
+      setImageLoaded(true);
+    };
+
+    if (imageElement.complete) {
+      handleLoad();
+    } else {
+      imageElement.addEventListener('load', handleLoad);
+    }
+
+    return () => {
+      imageElement.removeEventListener('load', handleLoad);
+    };
+  }, []);
+
+  return (
+    <div className={styles[`preview-${type}`]} title={filename}>
+      {onRemove ? (
+        isUploading || (type === 'image' && !imageLoaded) ? (
+          <FontAwesomeV6Icon
+            className={styles.topRightIcon}
+            iconName={'circle-notch'}
+            animationType={'spin'}
+          />
+        ) : (
+          <WithTooltip
+            tooltipProps={{
+              tooltipId: 'close-button',
+              direction: 'onTop',
+              size: 'xs',
+              text: aichatI18n.remove(),
+              className: styles.closeTooltip,
+            }}
+            tooltipOverlayClassName={styles.closeTooltipOverlay}
+          >
+            <button
+              className={classNames(
+                styles.topRightIcon,
+                styles.topRightIconClose
+              )}
+              type="button"
+              onClick={onRemove}
+            >
+              <FontAwesomeV6Icon iconName={'circle-xmark'} />
+            </button>
+          </WithTooltip>
+        )
+      ) : null}
+      {type === 'image' ? (
+        <div
+          className={!imageLoaded ? styles['preview-image-loading'] : undefined}
+        >
+          <img
```

Comment:
Mostly a copy from `StagedFilesPreview`, but main changes are moving to an `img` element rather than a image background via CSS and the logic to have a placeholder div while we wait for the image to upload (then download).

Reply:
One reason I had used the background image (rather than an img tag) was so the image would have rounded corners (going off of Mark's mocks). Do we still want that? Also noticed that the image is no longer cropped to a square - do we want to preserve that?

Reply:
There was a request not to crop images any more (asks are in the PR description) and here:
https://codedotorg.slack.com/archives/C06FELVTV0Q/p1742590944653719?thread_ts=1742412831.241269&cid=C06FELVTV0Q

Reply:
Re: rounding, I was thinking it was still rounding, but I'm zooming in and seeing that may not be the case. Was it not possible to round the image with an img element? The problem I was having with the image as background is it seemed to be tricky to get the container to match the size of the image when it was just a background

![image](https://github.com/user-attachments/assets/2d62fc73-5506-4fa8-8ee3-e42e5202ab7a)

![image](https://github.com/user-attachments/assets/1bb2ce0c-d37f-4b2a-bc0c-eb900bdf9713)

Reply:
Ahh my bad, totally missed that you already stated all this in the PR description 🤦 but yeah, I wasn't able to find a way to round the corners without using the image background. However, that was also an easier solution since the image had to be a fixed size anyway...I wonder if surrounding the image in a `div` and adding rounded corners to that would work?

Reply:
I put a border radius on the img element itself, so we're getting rounded edges there now at least:

![image](https://github.com/user-attachments/assets/1672dd9a-52e1-4e31-ac55-d6a41756c56e)

Reply:
Nice, yeah that looks good!

Reply:
Adding `overflow: hidden` to the outer div that has a border radius gives the desired result, without having to add an additional border radius to the image.

Reply:
hmm I tried that and was seeing the "remove" x button at the top right being mostly hidden as well. I'm sure there's a way to use overflow hidden though, CSS is not my forte...

Reply:
Yes, the trick is to create a common parent `div` for the X button and the image, with `position: relative`.  The X button can be rendered with `position: absolute` and the image can have the `div` with the border radius and the hidden overflow, and then the actual image inside of that.